odoo/models.py

```diff
@@ -2897,6 +2897,23 @@ def _field_to_sql(self, alias: str, fname: str, query: (Query | None) = None, fl
             if len(sql_field_langs) == 1:
                 return sql_field_langs[0]
             return SQL("COALESCE(%s)", SQL(", ").join(sql_field_langs))
+        if field.company_dependent:
+            default = field.get_company_dependent_fallback(self, format='column')
+            # in _read_group_orderby the result of field to sql will be mogrified and split to
+            # e.g SQL('COALESCE(%s->%s') and SQL('to_jsonb(%s))::boolean') as 2 orderby values
+            # and concatenated by SQL(',') in the final result, which works in an unexpected way
+            if field.type == 'boolean':
```

Comment:
This is not extensible. If this is the set of types we support, we should raise errors when someone declares a company_dependent field which is not one of those.

Reply:
We dont' want people extend it. There are so many attribute combinations doesn't work without warning. I think it is not a problem.

Reply:
I mean it should be refactored to avoid if-elif-elif-...